src/Symfony/Component/EventDispatcher/EventDispatcher.php

```diff
@@ -46,21 +44,23 @@ public function dispatch($eventName, Event $event = null)
         $event->setDispatcher($this);
         $event->setName($eventName);
 
-        if (!isset($this->listeners[$eventName])) {
```

Comment:
This doesn't bring much value and is not reliable anyway as it could also be set with an empty array by adding and removing a listener. By removing this we can also save a call to lazyLoad in the overwritten ContainerAwareEventDispatcher